lib/util/api-json.js

```diff
@@ -63,7 +63,11 @@ function api(token, endpoint, method, requestBody, callback) {
                 if (responseBody.trim().length === 0) {
                     responseBody = null;
                 } else {
-                    responseBody = JSON.parse(responseBody);
+                    try {
+                        responseBody = JSON.parse(responseBody);
+                    } catch (e) {
+                        return callback(new APIError(e, responseBody, responseCode));
```

Comment:
Also, does `APIError` work as expected when you pass in another `Error` object instead of a string?  From what I can tell, it's expecting a message, e.g. "Internal Server Error" or "Invalid credentials".